Guesser/PropelORMFieldGuesser.php

```diff
@@ -8,23 +8,34 @@
 
 class PropelORMFieldGuesser extends ContainerAware
 {
+    /**
+     * @var boolean
+     */
     private $guessRequired;
 
+    /**
+     * @var boolean
+     */
     private $defaultRequired;
 
+    /**
+     * @var array
+     */
     private $cache = array();
 
-    private static $current_class;
-
-    protected function getMetadatas($class = null)
+    /**
+     * @param $class
+     * @return mixed
+     */
+    protected function getMetadatas($class)
     {
-        if ($class) {
-            self::$current_class = $class;
-        }
-
-        return $this->getTable(self::$current_class);
+        return $this->getTable($class);
     }
 
+    /**
+     * @param $class
+     * @return array
+     */
     public function getAllFields($class)
     {
         $return = array();
```

Comment:
According to this: http://propelorm.org/documentation/cookbook/runtime-introspection.html... I'm not anymore sure :D but I think.

Experienced propel user are welcome to express themselves here